bin/yaml/c.yaml

```diff
@@ -60,13 +60,14 @@ compilers:
         untar_dir: sdcc-{name}
         targets:
           - 4.0.0
+          - 4.1.0
           - 4.2.0
           - 4.3.0
           - 4.4.0
       newer:
         untar_dir: sdcc
         targets:
-          - 4.1.0
+          - 4.5.0
```

Comment:
This won't cause the version to change here. We don't reinstall over directories every time; if the directory exists and the `check_exe` works, we assume all's well.

I'll take a look and try and work out what's going on here. Our current installations are:
```
 ls -ld /opt/compiler-explorer/sdcc*
drwxr-xr-x 4 ubuntu ubuntu 6144 Jan 30  2020 /opt/compiler-explorer/sdcc-4.0.0
drwxr-sr-x 4   1001   1001 6144 Mar  6  2021 /opt/compiler-explorer/sdcc-4.1.0
drwxr-xr-x 4 ubuntu ubuntu 6144 Mar  8  2022 /opt/compiler-explorer/sdcc-4.2.0
drwxr-xr-x 5 ubuntu ubuntu 6144 Jul  7  2023 /opt/compiler-explorer/sdcc-4.3.0
drwxr-xr-x 5 ubuntu ubuntu 6144 Jan 29  2024 /opt/compiler-explorer/sdcc-4.4.0
```